plugins/browser-plugin-youtube-tracking/src/index.ts

```diff
@@ -0,0 +1,231 @@
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+import { UrlParameters, trackingOptionsParser } from './helperFunctions';
+import { YTError, YTPlayerEvent, YTState, YTStateEvent } from './constants';
+import { EventData, MediaTrackingOptions, TrackedPlayer, TrackingOptions } from './types';
+import { BrowserPlugin, BrowserTracker, dispatchToTrackersInCollection } from '@snowplow/browser-tracker-core';
+import { buildSelfDescribingEvent, CommonEventProperties, Logger, SelfDescribingJson } from '@snowplow/tracker-core';
+import { SnowplowEvent } from './snowplowEvents';
+import { MediaPlayerEvent } from './contexts';
+import { buildYouTubeEvent } from './buildYouTubeEvent';
+import { YTEvent } from './youtubeEvents';
+
+declare global {
+  interface Window {
+    onYouTubeIframeAPIReady?: Function;
+  }
+}
+
+let LOG: Logger;
+export const _trackers: Record<string, BrowserTracker> = {};
+let tracked_players: Record<string, TrackedPlayer> = {};
+let iframeApiHasLoaded = false;
+let trackingQueue: Array<TrackingOptions> = [];
+
+export function YouTubeTrackingPlugin(): BrowserPlugin {
+  return {
+    activateBrowserPlugin: (tracker: BrowserTracker) => {
+      _trackers[tracker.id] = tracker;
+    },
+    logger: (logger) => {
+      LOG = logger;
+    },
+  };
+}
+
+function trackEvent(
+  event: SelfDescribingJson<MediaPlayerEvent> & CommonEventProperties,
+  trackers: Array<string> = Object.keys(_trackers)
+): void {
+  dispatchToTrackersInCollection(trackers, _trackers, (t) => {
+    t.core.track(buildSelfDescribingEvent({ event }), event.context, event.timestamp);
+  });
+}
+
+function enableYouTubeIframeAPI() {
+  const tag: HTMLScriptElement = document.createElement('script');
+  tag.src = 'https://www.youtube.com/iframe_api';
+  const firstScriptTag = document.getElementsByTagName('script')[0];
+  firstScriptTag.parentNode!.insertBefore(tag, firstScriptTag);
```

Comment:
Do we need to include this, or could we document this is a required prerequiste? I quite like the idea of including it, I'm just wondering if we need to, does it make it much nicer to use if this is a tracker function vs expecting users to add it themselves?